models/inh_poisson_generator.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ *  inh_poisson_generator.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "inh_poisson_generator.h"
+
+#include "event_delivery_manager_impl.h"
+#include "kernel_manager.h"
+
+#include "dict.h"
+#include "dictutils.h"
+#include "doubledatum.h"
+
+#include "exceptions.h"
+#include "integerdatum.h"
+#include "arraydatum.h"
+#include "numerics.h"
+#include "universal_data_logger_impl.h"
+
+#include <cmath>
+#include <limits>
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameter
+ * ---------------------------------------------------------------- */
+
+nest::inh_poisson_generator::Parameters_::Parameters_()
+  : rate_times_()  // ms
+  , rate_values_() // spikes/ms
+{
+}
+
+/* ----------------------------------------------------------------
+ * Parameter extraction and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::inh_poisson_generator::Parameters_::get( DictionaryDatum& d ) const
+{
+  ( *d )[ names::rate_times ] =
+    DoubleVectorDatum( new std::vector< double_t >( rate_times_ ) );
+  ( *d )[ names::rate_values ] =
+    DoubleVectorDatum( new std::vector< double_t >( rate_values_ ) );
+}
+
+void
+nest::inh_poisson_generator::Parameters_::set( const DictionaryDatum& d,
+  Buffers_& b )
+{
+  const bool ut =
+    updateValue< std::vector< double_t > >( d, names::rate_times, rate_times_ );
+  const bool uv = updateValue< std::vector< double_t > >(
+    d, names::rate_values, rate_values_ );
```

Comment:
`ut` and `uv` are rather too short and uninformative.

Reply:
done